enhancements/single-node/aos-sno-pair.md

```diff
@@ -0,0 +1,143 @@
+
+Where this is not possible, there are existing capabilities that we can leverage by enabling the _Corosync two_node, last_man_standing, and wait_for_all_ options (See: man votequorum) and configuring a fencing delay in Pacemaker.
+
+In combination, these options provide the following behaviour:
+- Both sides retain quorum in the event of a network split.
+- Service recovery is normally predicated on successful fencing.
+- When the fenced node reboots, it will not be able to obtain quorum until it has seen its peer (preventing an infinite fencing cycle).
+- To avoid both sides fencing each other at the same time, fencing is typically configured to occur after a delay on one of the nodes. That node will perform recovery in the event of a true peer failure, but will always lose in the response to a pure network failure.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Open Questions
+
+- Do we really need active/active, or could active/passive be sufficient?
+  - Reasons mentioned in favor:
+    - It ensures the secondary node works when you need it. Not only it is working, but the whole list of dependencies it carries with it (DNS...)
+    - Performance.
+    - Cost. The secondary node is not just sitting there, wasting resources.
+    - LESS maintenance procedures. You don't need to failover/failback on upgrades, for example.
+- Do we really need automated failover, or is manual confirmation sufficient?
+- How much overlap is there between the goals as laid out here and Telco RAN requirements?
+
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Examples
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+## Implementation History
+
+## Drawbacks
+This approach introduces a non-kubernetes HA approach to an OpenShift environment, requiring users to understand both kubernetes and RHEL-HA.
+
+## Alternatives
+
+1. Support an active/passive configuration with 1 master + 1 worker.  Workloads stay running on the worker if the master dies (unless they require rescheduling before the master is recovered), but the master takes over if the worker dies.
+2. Run with a single active etcd copy, but make use of [etcd learners](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/design-learner.md) to provide a non-live replica that can be promoted in the event the active copy fails.
+   Learners will gain self-promotion abilities for automated failover in etcd 3.5.  Would require additional engineering (either in etcd or some component driving it) for the learner to distinguish between a network and leader failures, and/or ensure the leader is no longer accepting writes before promoting itself.
```

Comment:
```suggestion
   Learners will gain self-promotion abilities for automated failover in etcd 3.5.  Would require additional engineering (either in etcd or some component driving it) for the learner to distinguish between a network and leader failures, and/or ensure the leader is no longer accepting writes before promoting itself.
   Strictly speaking both sides should loose quorum and the cluster would be hosed. If a winner is pre-ordained (eg. the leader), there is a significant chance that's the one that will fail and the cluster is still hosed. If both sides are forced to retain quorum, you still need a mechanism to prevent volumes being mounted, and StatefulSets from being started, on both sides.
```